From 4314c66376672e0f1cff13106ba9c46e081fdfc6 Mon Sep 17 00:00:00 2001
From: Hamster Tian <haotia@gmail.com>
Date: Tue, 19 Jun 2018 23:55:14 +0800
Subject: [PATCH 37/40] SystemUI: tune battery level ranges for Bluetooth
 statusbar icon

* We have icons for 5 levels: 0%, 25%, 50%, 75% and 100%.
  Battery levels reported by peripheral should be indicated by
  the closest one. That is to say, the icons should represent
  following level:

  0%:   (0%,    12.5%)
  25%:  [12.5%, 37.5%)
  50%:  [37.5%, 62.5%)
  75%:  [62.5%, 87.5%)
  100%: [87.5%, 100%]

  This makes more sense than original ranges. A real-life
  example is Sony headsets: they only report 4 levels 20% 50%
  70% and 100%. Corresponding icons would be 0%, 50%, 50%, 100%
  with original values, and 25% 50% 75% 100% with this commit.
  The latter is obviously closer to real battery level.

Change-Id: Ie72ab7c40caff3443d71734b3ca69b2f43119053
---
 .../systemui/statusbar/phone/PhoneStatusBarPolicy.java    | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/phone/PhoneStatusBarPolicy.java b/packages/SystemUI/src/com/android/systemui/statusbar/phone/PhoneStatusBarPolicy.java
index 5d4ed7fa60e..0dca56b1341 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/phone/PhoneStatusBarPolicy.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/phone/PhoneStatusBarPolicy.java
@@ -420,13 +420,13 @@ public class PhoneStatusBarPolicy implements Callback, Callbacks,
             bluetoothEnabled = mBluetooth.isBluetoothEnabled();
             if (mBluetooth.isBluetoothConnected()) {
                 int batteryLevel = mBluetooth.getLastDevice().getBatteryLevel();
-                if (batteryLevel == 100) {
+                if (batteryLevel >= 88) {
                     iconId = R.drawable.stat_sys_data_bluetooth_connected_battery_5;
-                } else if (batteryLevel > 75) {
+                } else if (batteryLevel >= 63) {
                     iconId = R.drawable.stat_sys_data_bluetooth_connected_battery_4;
-                } else if (batteryLevel > 50) {
+                } else if (batteryLevel >= 38) {
                     iconId = R.drawable.stat_sys_data_bluetooth_connected_battery_3;
-                } else if (batteryLevel > 25) {
+                } else if (batteryLevel >= 13) {
                     iconId = R.drawable.stat_sys_data_bluetooth_connected_battery_2;
                 } else if (batteryLevel > 0) {
                     iconId = R.drawable.stat_sys_data_bluetooth_connected_battery_1;
-- 
2.17.1

